evap/staff/views.py

```diff
@@ -144,159 +149,192 @@ def __init__(self):
     return render(request, "staff_semester_view.html", template_data)
 
 
+class EvaluationOperation:
+    email_template_name = None
+    confirmation_message = None
+
+    @staticmethod
+    def applicable_to(evaluation):
+        raise NotImplementedError
+
+    @staticmethod
+    def warning_for_inapplicables(amount):
+        raise NotImplementedError
+
+    @staticmethod
+    def apply(request, evaluations, email_template=None):
+        raise NotImplementedError
+
+
+class RevertToNewOperation(EvaluationOperation):
+    confirmation_message = ugettext_lazy("Do you want to revert the following evaluations to preparation?")
+
+    @staticmethod
+    def applicable_to(evaluation):
+        return evaluation.state in ['prepared', 'editor_approved', 'approved']
+
+    @staticmethod
+    def warning_for_inapplicables(amount):
+        return ungettext("{} evaluation can not be reverted, because it already started. It was removed from the selection.",
+            "{} evaluations can not be reverted, because they already started. They were removed from the selection.", amount).format(amount)
+
+    @staticmethod
+    def apply(request, evaluations, email_template=None):
+        for evaluation in evaluations:
+            evaluation.revert_to_new()
+            evaluation.save()
+        messages.success(request, ungettext("Successfully reverted {} evaluation to in preparation.",
+            "Successfully reverted {} evaluations to in preparation.", len(evaluations)).format(len(evaluations)))
+
+
+class RevertToPreparedOperation(EvaluationOperation):
+    email_template_name = EmailTemplate.EDITOR_REVIEW_NOTICE
+    confirmation_message = ugettext_lazy("Do you want to revert the following evaluations to preparation?")
+
+    @staticmethod
+    def applicable_to(evaluation):
+        return evaluation.state in ['new', 'editor_approved']
+
+    @staticmethod
+    def warning_for_inapplicables(amount):
+        return ungettext("{} evaluation can not be reverted, because it already started. It was removed from the selection.",
+            "{} evaluations can not be reverted, because they already started. They were removed from the selection.", amount).format(amount)
+
+    @staticmethod
+    def apply(request, evaluations, email_template=None):
+        for evaluation in evaluations:
+            evaluation.ready_for_editors()
```

Comment:
the methods should optimally use the same names as the states. it is a bit confusing that `ready_for_editors` moves the evaluation to `prepared`. maybe `move_to_...` (`move_from_..._to_...` if the direction is important).

Reply:
There is a name mismatch with

* `MoveToPreparedOperation` and `evaluation.ready_for_editors`
* `StartEvaluationOperation` and `evaluation.evaluation_begin`
* `RevertToReviewedOperation` and `evaluation.unpublish()`

Do you agree that there should be another refactoring for the state transition names in the model to make it more consistent among themselves and throughout the whole project.

Reply:
yes. would you open an issue for that?